### ISSUE TITLE
Missing types and possibly other core files in 6.0.9

### ISSUE THREAD

**Issue by ZLJasonG** [styled-components/styled-components#4191]
## Steps to reproduce
npm install styled-components

import { ThemeProvider } from "styled-components"

## Expected Behavior
File imports without missing types error

## Actual Behavior
TS7016: Could not find a declaration file for module 'styled-components'. '/node_modules/styled-components/dist/styled-components.cjs.js' implicitly has an 'any' type.

It looks like many files have been dropped from the dist folder of the NPM deployment of 6.0.9 when compared to 6.0.8 including all the .d.ts files
https://www.npmjs.com/package/styled-components/v/6.0.8?activeTab=code
https://www.npmjs.com/package/styled-components/v/6.0.9?activeTab=code

**Comment by MonkSoul** [comment]
same problem~

**Comment by Rechdan** [comment]
Same.

**Comment by cabbage-cart** [comment]
same.

**Comment by zhenzhenChange** [comment]
same.

**Comment by rigobauer** [comment]
Same

**Comment by Pouchey** [comment]
same

**Comment by nyxb** [comment]
me too!

```
Could not find a declaration file for module 'styled-components'. '/Users/nyxb/projects/websites/nyxb.gives/node_modules/styled-components/dist/styled-components.cjs.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/styled-components` if it exists or add a new declaration (.d.ts) file containing `declare module 'styled-components';`ts(7016)
Could not find a declaration file for module 'styled-components'. '/Users/nyxb/projects/websites/nyxb.gives/node_modules/styled-components/dist/styled-components.cjs.js' implicitly has an 'any' type.
```

**Comment by victazzz** [comment]
same

**Comment by andelizondo** [comment]
I noticed that the `dist` folder is almost empty on version `6.0.9`:

![Screenshot 2023-10-13 at 14 35 14](https://github.com/styled-components/styled-components/assets/7799054/a0f89a54-ea59-44f0-9ef6-cd99e4fcfc94)

whereas on version `6.0.8` the `dist` folder contains the correct `types.d.ts` file needed:

![Screenshot 2023-10-13 at 14 35 48](https://github.com/styled-components/styled-components/assets/7799054/faa481fa-93ab-4b5a-8a6a-e8594701abd9)

**Comment by ZLJasonG** [comment]
Looks to be resolved in 6.1.0 :) Thanks
